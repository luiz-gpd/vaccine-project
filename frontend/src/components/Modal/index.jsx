import React from 'react'
import ClayButton from '@clayui/button'
import ClayModal from '@clayui/modal'
import { ClayInput } from '@clayui/form'
import api from '../../utils/api'

const Modal = ({
    visible,
    onClose,
    observer,
    title,
    id,
    children
}) => {

    const initialState = {
        information:""
    }
    const [showChanger, setShowChanger] = React.useState(false)
    const [form, setForm] = React.useState(initialState)
    
    const onChange = ({ target: { name, value } }) => {
        setForm({
          ...form,
          [name]: value,
        });
      };

    const onClick = async () => {
        const formData = {
            consultInfo:form.information
        }
        await api.put(`/user/${id}`, formData);
        document.location.reload();
    }

    return (
        <>
            {visible && (
                <ClayModal
                    observer={observer}
                    size="lg"
                    status="info"
                >
                    <ClayModal.Header>{title}</ClayModal.Header>
                    <ClayModal.Body>
                        {children}
                    </ClayModal.Body>
                    <ClayModal.Footer
                        first={<>
                                <ClayButton displayType="secondary" onClick={setShowChanger}>Alterar</ClayButton>
                                {showChanger && (<><ClayInput data-testid='formField' name="information" className="mt-4 mb-2 formfield" value={form.information} onChange={onChange}/>
                                <ClayButton displayType="primary" onClick={onClick}>Salvar</ClayButton></>)}
                        </>}
                        last={<ClayButton data-testid='closeButtonSave' className="btn btn-danger" onClick={onClose}>Fechar</ClayButton>}
                    />
                </ClayModal>
            )}
        </>
    );
}

export default Modal;
