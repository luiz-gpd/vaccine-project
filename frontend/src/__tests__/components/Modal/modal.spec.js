import Modal from '../../../components/Modal'

import { render, fireEvent } from "@testing-library/react";

describe("Modal component", () => {
    it("renders", () => {
        const { asFragment } = render(<Modal />)

        expect(asFragment()).toMatchSnapshot();
    });

    it("renders with context data", () => {
        const { queryByText } = render(
            <Modal visible={false}
            // onClose=
            // observer=
            title="modal1"
            id="modal1"
            >Tudo bem</Modal>
        );

        // const savingButton = queryByText("Salvar");

        // fireEvent.click(savingButton)

    })
});