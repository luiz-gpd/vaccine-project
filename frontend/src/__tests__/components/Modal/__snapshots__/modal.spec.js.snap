// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders 1`] = `<DocumentFragment />`;
